nioChecksumTest/src/main/scala/cromwell/niochecksumtest/NioChecksumTest.scala

```diff
@@ -0,0 +1,123 @@
+  * (5) Update `mySecretFile` below to point to the local adc.json file for your Google account
+  * (6) "Run" NioChecksumTest using IntelliJ
+  * (7) Stop NioChecksumTest using IntelliJ because it does not stop on its own :(
+  */
+object NioChecksumTest {
+
+  /////////////////////
+  // UPDATE THESE
+  // V V V V V
+  private val marthaUrl = "https://us-central1-broad-dsde-dev.cloudfunctions.net/martha_v3"
+  private val drsPathToResolve = "drs://dg.4DFC:011a6a54-1bfe-4df9-ae24-990b12a812d3"
+  private val mySecretFile = "/Users/[USER]/.config/gcloud/legacy_credentials/[GOOGLE_ACCOUNT_NAME]/adc.json"
+  // ^ ^ ^ ^ ^
+  /////////////////////
+
+  private lazy val systemConfig = ConfigFactory.load()
+
+  private lazy val drsConfig = ConfigFactory.parseString(
+    s"""martha.url = "$marthaUrl"
+      | access-token-acceptable-ttl = 1 hour
+      |""".stripMargin
+  )
+
+  private lazy val authMode = UserMode("user auth", mySecretFile)
+  private lazy val credentials = GoogleDrsCredentials(
+    authMode.credentials(
+      // From GoogleAccessTokenStrategy in CromwellDrsLocalizer
+      List(
+        "https://www.googleapis.com/auth/userinfo.email",
+        "https://www.googleapis.com/auth/userinfo.profile"
+      )
+    ),
+    drsConfig
+  )
+
+  def main(args: Array[String]): Unit = {
+    implicit val context: ActorSystem = ActorSystem("NioChecksumTest")
+    implicit val materializer: ActorMaterializer = ActorMaterializer()
+    val serviceRegistryActor: ActorRef = context.actorOf(
+      ServiceRegistryActor.props(systemConfig),
+      name = "NioChecksumTestServiceRegistryActor"
+    )
+
+    val ioActor: ActorRef = context.actorOf(
+      IoActor.props(
+        ioConfig = systemConfig.as[IoConfig],
+        serviceRegistryActor = serviceRegistryActor,
+        applicationName = "NioChecksumTestIoActor"),
+      name = "IoActor"
+    )
+
+    val ioPromiseProxyActor = context.actorOf(
+      IoPromiseProxyActor.props(ioActor),
+      name = "IoPromiseProxyActor"
+    )
+
+    try {
+      println(resolveDrsFile(
+        drsPathToResolve,
+        new AsyncIo(ioPromiseProxyActor, IoCommandBuilder())
+      ))
+    } finally {
+      context.stop(ioPromiseProxyActor)
+      context.stop(ioActor)
+      context.stop(serviceRegistryActor)
+    }
+
+    // Compiler gets mad if we don't use the result of the terminate call.
+    println(s"Termination result is ${Await.result(context.terminate(), 60.seconds)}")
+  }
+
+  def resolveDrsFile(pathToResolve: String, asyncIo: AsyncIo): String = {
+
+    val drsFilesystemProvider = new DrsCloudNioFileSystemProvider(
+      drsConfig,
+      credentials,
+      DrsReader.readInterpreter(Option(authMode), WorkflowOptions(JsObject.empty), None)
+    )
+
+    val drsPath = DrsPath(drsFilesystemProvider.getCloudNioPath(pathToResolve), None)
+
+    val fileContents = Try(
+      Await.result(
+        asyncIo.contentAsStringAsync(drsPath, Option(100000), failOnOverflow = false),
```

Comment:
should this perhaps fail on overflow so we're not wondering why checksums don't match? 😄

Reply:
Yes, yes it should. I'm also realizing now that I randomly stuck that `Option(100000)` in there and never went back to think about what a reasonable value should be. Any recommendations?

Reply:
That's probably fine, I'm assuming we don't need huge files for what we're trying to do with this program.